substrate/bin/node/runtime/src/genesis_config_presets.rs

```diff
@@ -0,0 +1,227 @@
+
+/// Helper function to create RuntimeGenesisConfig json patch for testing.
+pub fn kitchen_sink_genesis(
+	initial_authorities: Vec<(AccountId, AccountId, SessionKeys)>,
+	root_key: AccountId,
+	endowed_accounts: Vec<AccountId>,
+	stakers: Vec<Staker>,
+	staking_playground_config: Option<StakingPlaygroundConfig>,
+) -> serde_json::Value {
+	let (validator_count, min_validator_count, dev_stakers) = match staking_playground_config {
+		Some(c) => (c.validator_count, c.minimum_validator_count, Some(c.dev_stakers)),
+		None => {
+			let authorities_count = initial_authorities.len() as u32;
+			(authorities_count, authorities_count, None)
+		},
+	};
+
+	let collective = collective(&endowed_accounts);
+
+	build_struct_json_patch!(RuntimeGenesisConfig {
+		balances: BalancesConfig {
+			balances: endowed_accounts.iter().cloned().map(|x| (x, ENDOWMENT)).collect(),
+			..Default::default()
+		},
+		session: SessionConfig {
+			keys: initial_authorities
+				.iter()
+				.map(|x| { (x.0.clone(), x.1.clone(), x.2.clone()) })
+				.collect(),
+		},
+		staking: StakingConfig {
+			validator_count,
+			minimum_validator_count: min_validator_count,
+			invulnerables: initial_authorities
+				.iter()
+				.map(|x| x.0.clone())
+				.collect::<Vec<_>>()
+				.try_into()
+				.expect("too many authorities"),
+			slash_reward_fraction: Perbill::from_percent(10),
+			stakers,
+			dev_stakers
+		},
+		elections: ElectionsConfig {
+			members: collective.iter().cloned().map(|member| (member, STASH)).collect(),
+		},
+		technical_committee: TechnicalCommitteeConfig { members: collective },
+		sudo: SudoConfig { key: Some(root_key) },
+		babe: BabeConfig { epoch_config: BABE_GENESIS_EPOCH_CONFIG },
+		society: SocietyConfig { pot: 0 },
+		assets: AssetsConfig {
+			// This asset is used by the NIS pallet as counterpart currency.
+			assets: vec![(9, Sr25519Keyring::Alice.to_account_id(), true, 1)],
+			..Default::default()
+		},
+		nomination_pools: NominationPoolsConfig {
+			min_create_bond: 10 * DOLLARS,
+			min_join_bond: 1 * DOLLARS,
+		},
+	})
+}
+
+/// Provides the JSON representation of predefined genesis config for given `id`.
+pub fn get_preset(id: &PresetId) -> Option<Vec<u8>> {
+	// Note: Can't use `Sr25519Keyring::Alice.to_seed()` because the seed comes with `//`.
+	let (alice_stash, alice, alice_session_keys) = authority_keys_from_seed("Alice");
+	let (bob_stash, _bob, bob_session_keys) = authority_keys_from_seed(&"Bob");
+
+	let endowed = well_known_including_eth_accounts();
+
+	let patch = match id.as_ref() {
+		sp_genesis_builder::DEV_RUNTIME_PRESET => kitchen_sink_genesis(
+			vec![(alice_stash.clone(), alice_stash.clone(), alice_session_keys)],
+			alice.clone(),
+			endowed,
+			vec![validator(alice_stash.clone())],
+			None,
+		),
+		sp_genesis_builder::LOCAL_TESTNET_RUNTIME_PRESET => kitchen_sink_genesis(
+			vec![
+				(alice_stash.clone(), alice_stash.clone(), alice_session_keys),
```

Comment:
Using `alice_stash` as the controller account preserves the previous behavior. So I thought I will keep it like that.

Reply:
Why this was change to `alice_stash`? Would you plz drop some comment on this?

Reply:
This is already how it was, but much better hidden 🙈, https://github.com/paritytech/polkadot-sdk/blob/1bc6ca606438a65c927f14be3f36634ca0e58e8f/substrate/bin/node/cli/src/chain_spec.rs#L384.

I can imagine that is just simpler for testing if the stash account is also the controller account. I don't really have a preference here. I can try changing this to the controller account and see if it still works.

Reply:
Ok, grandpa doesn't seem to like when there are controller accounts not equal to the stash accounts at genesis. So I keep it as the stash account and made a comment. 👍 

Thanks for the very detailed review! 🚀